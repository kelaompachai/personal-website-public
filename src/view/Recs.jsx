import React, { useState } from 'react';
import IndividualRec from './IndividualRec';
import recsList from '../model/recommendations';
import CarouselButton from './CarouselButton';

function Recs() {
  const [showing, setShowing] = useState(0);

  const recsArray = recsList.map(
    (el) => <IndividualRec giver={el.giver} relation={el.relation} text={el.text} />
  );

  function carousel(direction) {
    if (direction === 'right') setShowing((showing + 1) % recsArray.length);
    else setShowing((showing + recsArray.length - 1) % recsArray.length);

    const elements = document.querySelectorAll('.cheese');

    elements.forEach((el) => {
      el.classList.remove('animate-blurin');
    });

    const cheese = elements[0].offsetWidth;

    elements.forEach((el) => {
      el.classList.add('animate-blurin');
    });
  }

  return (
    <div className="flex items-center justify-around" >
      <CarouselButton direction="left" carousel={() => { carousel('left'); }} />
      <div className="flex items-center justify-center">
        {recsArray[showing]}
      </div>
      <CarouselButton direction="right" carousel={() => { carousel('right'); }} />
    </div>
  );
}

export default Recs;
