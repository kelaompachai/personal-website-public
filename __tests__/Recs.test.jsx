import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Recs from '../src/view/Recs';


describe('Recs component renders properly', () => {
  it('matches snapshot and changes content when the button is pushed', async () => {
    expect(render(<Recs />).baseElement).toMatchSnapshot();

    const btn = screen.getAllByRole('button')[0];
    expect(btn).not.toBeDisabled();

    expect(screen.queryByText(/Joe was a great asset to our team/)).toBeInTheDocument();

    await userEvent.click(btn);

    expect(screen.queryByText(/Joe was a great asset to our team/)).not.toBeInTheDocument();
  });
});
