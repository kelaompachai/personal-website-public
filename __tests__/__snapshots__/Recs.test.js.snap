// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recs component renders properly matches snapshot and changes content when the button is pushed 1`] = `
<body>
  <div>
    <div
      class="flex items-center justify-around"
    >
      <button
        class="flex justify-center items-center size-24 border-2 border-well-read-500 rounded-3xl bg-well-read-200 hover:bg-well-read-300 active:transform active:translate-y-0.5"
        type="button"
      >
        <svg
          class="size-1/2 fill-well-read-500"
          viewBox="0 0 16 16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15 8a.5.5 0 0 0-.5-.5H2.707l3.147-3.146a.5.5 0 1 0-.708-.708l-4 4a.5.5 0 0 0 0 .708l4 4a.5.5 0 0 0 .708-.708L2.707 8.5H14.5A.5.5 0 0 0 15 8"
            fill-rule="evenodd"
          />
        </svg>
      </button>
      <div
        class="flex items-center justify-center"
      >
        <div
          class="flex justify-center items-center min-h-72 my-2.5"
        >
          <div
            class="h-[26rem] flex flex-col max-w-xl bg-well-read-200 rounded-3xl"
          >
            <div
              class="flex items-center justify-center"
            >
              <div
                class="cheese flex-grow-0 mx-5 text-gray-800 font-medium"
              >
                Andrew Cirt
              </div>
              <div
                class="cheese flex-grow-0 mx-5 text-gray-800 font-light"
              >
                Colleague at KafOpticon
              </div>
            </div>
            <div
              class="flex items-center justify-center flex-grow"
            >
              <p
                class="cheese p-5 text-center text-gray-600 font-light"
              >
                Joe was a great asset to our team, both from a technical and non-technical perspective. Whether he was making quick and significant progress on his tasks or helping others make progress on theirs, Joe was always willing to help and go the extra mile. Joe provided valuable insight into solving problems and bugs and also communicated with others about solutions. Joe has been great to work with and I'm sure he will continue to be so for others.
              </p>
            </div>
          </div>
        </div>
      </div>
      <button
        class="flex justify-center items-center size-24 border-2 border-well-read-500 rounded-3xl bg-well-read-200 hover:bg-well-read-300 active:transform active:translate-y-0.5"
        type="button"
      >
        <svg
          class="size-1/2 fill-well-read-500"
          viewBox="0 0 16 16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 8a.5.5 0 0 1 .5-.5h11.793l-3.147-3.146a.5.5 0 0 1 .708-.708l4 4a.5.5 0 0 1 0 .708l-4 4a.5.5 0 0 1-.708-.708L13.293 8.5H1.5A.5.5 0 0 1 1 8"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </div>
  </div>
</body>
`;
